src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -91,5 +91,25 @@ public static long SwapLong(long value)
             Array.Reverse(bytes);
             return BitConverter.ToInt64(bytes, 0);
         }
+
+        /// <summary>
+        /// Toggles the endianness of the specified 32-bit integer.
+        /// </summary>
+        public static int SwapInt(int value)
+        {
+            byte[] bytes = BitConverter.GetBytes(value);
```

Comment:
please replace with in-place bit shift-based operations, Check http://referencesource.microsoft.com/#System/net/System/Net/IPAddress.cs,aa84b64c78670092 (#else part)